artnet.py

```diff
@@ -58,26 +58,28 @@ def __init__(self, ip_addr, *args, **kwargs):
         self._socket = socket(AF_INET, SOCK_DGRAM)
         self._socket.bind((ip_addr, ARTNET_PORT))
         self._socket.settimeout(30)
-        self._stop = False
+        self._stopped = False
 
     def stop(self):
-        self._socket.shutdown(SHUT_RDWR)
-        self._stop = True           
+        try:
+            self._socket.shutdown(SHUT_RDWR)
+        except Exception as e:
+            print(e)
+        self._stopped = True
 
     def run(self):
-        while not self._stop:
+        while not self._stopped:
             try:
                 data, _ = self._socket.recvfrom(1024)
                 packet = ArtnetPacket.build(data)
                 self._dmx.artnet_status = 'online'
 
                 if (packet.universe >= len(self._dmx.universes)):
                     continue
-                if (self._dmx.universes[packet.universe].input != 'ARTNET'):
+                if (not self._dmx.universes[packet.universe] or self._dmx.universes[packet.universe].input != 'ARTNET'):
```

Comment:
```suggestion
                if (not self._dmx.universes[packet.universe]):
                    continue
                if (self._dmx.universes[packet.universe].input != 'ARTNET'):
```
Keeping a single check for early-return `ifs` adheres better to the project code style.